EasyPost.Tests/UtilitiesTests/JsonSerializationTest.cs

```diff
@@ -0,0 +1,150 @@
+    {
+        #region Tests
+
+        [Fact]
+        [Testing.Function]
+        public void TestConvertJsonToDictionary()
+        {
+            try
+            {
+                JsonSerialization.ConvertJsonToObject<Dictionary<string, object>>(Json);
+            }
+            catch (Exception)
+            {
+                Assert.Fail("Failed to deserialize JSON");
+            }
+
+            // bad JSON data should throw an exception
+            Assert.Throws<JsonDeserializationError>(() => JsonSerialization.ConvertJsonToObject<Dictionary<string, object>>("bad_json_data"));
+
+            // missing data should throw an exception
+            Assert.Throws<JsonNoDataError>(() => JsonSerialization.ConvertJsonToObject<Dictionary<string, object>>(data: null));
+        }
+
+        [Fact]
+        [Testing.Function]
+        public void TestConvertJsonToExpandoObject()
+        {
+            try
+            {
+                JsonSerialization.ConvertJsonToObject(Json);
+            }
+            catch (Exception)
+            {
+                Assert.Fail("Failed to deserialize JSON");
+            }
+
+            // bad JSON data should throw an exception
+            Assert.Throws<JsonDeserializationError>(() => JsonSerialization.ConvertJsonToObject("bad_json_data"));
+
+            // missing data should throw an exception
+            Assert.Throws<JsonNoDataError>(() => JsonSerialization.ConvertJsonToObject<JsonObjectDetails>(data: null));
+        }
+
+        [Fact]
+        [Testing.Function]
+        public void TestConvertJsonToObject()
+        {
+            try
+            {
+                JsonSerialization.ConvertJsonToObject<JsonObjectRoot>(Json);
+            }
+            catch (Exception)
+            {
+                Assert.Fail("Failed to deserialize JSON");
+            }
+
+            // bad JSON data should throw an exception
+            Assert.Throws<JsonDeserializationError>(() => JsonSerialization.ConvertJsonToObject<JsonObjectRoot>("bad_json_data"));
+
+            // missing data should throw an exception
+            Assert.Throws<JsonNoDataError>(() => JsonSerialization.ConvertJsonToObject<JsonObjectRoot>(data: null));
+        }
+
+        [Fact]
+        [Testing.Function]
+        public void TestConvertObjectToJson()
+        {
+            try
+            {
+                JsonObjectRoot root = new()
+                {
+                    Id = "id",
+                    Details = new JsonObjectDetails
+                    {
+                        Name = "details_name",
+                        Value = 100
+                    }
+                };
+
+                JsonSerialization.ConvertObjectToJson(root);
+            }
+            catch (Exception)
+            {
+                Assert.Fail("Failed to serialize JSON");
+            }
+
+            // bad object should throw an exception (in theory, can't actually trigger this, Newtonsoft.Json is too smart)
```

Comment:
This comment needs to move up

Reply:
Just clarifying to future travelers why the unit test is this way, and why there is no really good assertion